src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt

```diff
@@ -82,67 +83,108 @@ class UserReceiptMailBuilder(@Autowired private val confidentialDataUtils: Confi
-          TransactionTemplate(
-            baseTransactionWithRequestedUserReceipt.transactionId
-              .value()
-              .toString()
-              .lowercase(Locale.getDefault()),
-            dateTimeToHumanReadableString(
-              ZonedDateTime.parse(transactionUserReceiptData.paymentDate),
-              Locale.forLanguageTag(language)),
-            amountToHumanReadableString(
-              baseTransactionWithRequestedUserReceipt.paymentNotices
-                .stream()
-                .mapToInt { paymentNotice: PaymentNotice ->
-                  paymentNotice.transactionAmount().value()
-                }
-                .sum() + transactionAuthorizationRequestData.fee),
-            PspTemplate(
-              transactionAuthorizationRequestData.pspBusinessName,
-              FeeTemplate(amountToHumanReadableString(transactionAuthorizationRequestData.fee))),
-            transactionAuthorizationCompletedData.rrn
-              ?: transactionAuthorizationRequestData.authorizationRequestId,
-            // TODO: auth code is mandatory for OK mail template but not received for PAYPAL
-            // TODO task CHK-1970 will take cares of handle mail template for APM methods
-            transactionAuthorizationCompletedData.authorizationCode ?: "-",
-            PaymentMethodTemplate(
-              transactionAuthorizationRequestData.paymentMethodDescription,
-              transactionAuthorizationRequestData.transactionGatewayAuthorizationRequestedData.logo
-                .toString(),
+        when (transactionAuthorizationRequestData.paymentTypeCode) {
+          "PPAL",
+          "MyBank" ->
+            createSuccessTemplate(
+              baseTransactionWithRequestedUserReceipt = baseTransactionWithRequestedUserReceipt,
+              emailAddress = emailAddress,
+              authorizationCode = "-")
+          "BPAY" -> {
+            val transactionGatewayAuthorizationData =
+              transactionAuthorizationCompletedData.transactionGatewayAuthorizationData
+            if (transactionGatewayAuthorizationData is NpgTransactionGatewayAuthorizationData) {
+              createSuccessTemplate(
+                baseTransactionWithRequestedUserReceipt = baseTransactionWithRequestedUserReceipt,
+                emailAddress = emailAddress,
+                authorizationCode = transactionGatewayAuthorizationData.operationId)
+            } else {
+              throw RuntimeException(
+                "Unexpected TransactionGatewayAuthorization for paymentTypeCode ${transactionAuthorizationRequestData.paymentTypeCode}. Expected NPG gateway.")
+            }
+          }
+          "Redirect" ->
+            createSuccessTemplate(
+              baseTransactionWithRequestedUserReceipt = baseTransactionWithRequestedUserReceipt,
+              emailAddress = emailAddress,
+              authorizationCode = transactionAuthorizationCompletedData.authorizationCode ?: "-")
+          else -> {
```

Comment:
What cases are covered by this `else`?

Reply:
I had a mistaken understanding of the data that the paymentTypeCode could take :')